roles/confluent.control_center/molecule/default/verify.yml

```diff
@@ -0,0 +1,162 @@
+---
+- name: Verify - kafka_broker
+  hosts: kafka_broker
+  gather_facts: no
+  tasks:
+    - name: Check line in server.properties
+      shell: |
+        grep -Fq 'ldap.java.naming.provider.url=ldap://ldap1:389' /etc/kafka/server.properties
+      register: linecheck
+      check_mode: no
+      changed_when: no
+      failed_when: linecheck.rc != 0
+
+- name: Verify - schema_registry
+  hosts: schema_registry
+  gather_facts: no
+  tasks:
+    - name: Check line properties file
+      shell: |
+        grep -Fq 'public.key.path=/var/ssl/private/public.pem' /etc/schema-registry/schema-registry.properties
+      register: linecheck
+      check_mode: no
+      changed_when: no
+      failed_when: linecheck.rc != 0
+
+- name: Verify - kafka_rest
+  hosts: kafka_rest
+  gather_facts: no
+  tasks:
+    - name: Check line properties file
+      shell: |
+        grep -Fq 'public.key.path=/var/ssl/private/public.pem' /etc/kafka-rest/kafka-rest.properties
+      register: linecheck
+      check_mode: no
+      changed_when: no
+      failed_when: linecheck.rc != 0
+
+- name: Verify - kafka_connect
+  hosts: kafka_connect
+  gather_facts: no
+  tasks:
+    - name: Check line properties file
+      shell: |
+        grep -Fq 'public.key.path=/var/ssl/private/public.pem' /etc/kafka/connect-distributed.properties
+      register: linecheck
+      check_mode: no
+      changed_when: no
+      failed_when: linecheck.rc != 0
+
+    # Below tests are for connect secret registry
+    - name: Get Connect Cluster Details
+      uri:
+        url: https://localhost:8083
+        status_code: 200
+        validate_certs: no
+        url_username: kafka-connect1
+        url_password: password
+        force_basic_auth: true
+      register: connect_cluster_query
+
+    - set_fact:
+        connect_cluster_json: "{{connect_cluster_query.json}}"
+
+    - name: Set kafka_cluster_id Variable
+      set_fact:
+        kafka_cluster_id: "{{connect_cluster_json.kafka_cluster_id}}"
+
+    - name: Grant role System Admin to mds user on Connect Cluster and Kafka Cluster
+      shell: |
+        curl -X POST https://kafka-broker1:8090/security/1.0/principals/User:mds/roles/SystemAdmin \
+          -u "mds":"password" -H "Content-Type: application/json" \
+          --cacert /var/ssl/private/ca.crt --key /var/ssl/private/kafka_connect.key --cert /var/ssl/private/kafka_connect.crt \
+          -d '{"clusters":{"kafka-cluster":"{{kafka_cluster_id}}","connect-cluster":"connect-cluster"}}'
+
+    # TODO add another user in molecule.yml and use this user instead of mds
```

Comment:
"The SystemAdmin role is overkill here; we should split this up into a higher number of less-permissive role bindings. I think we can do this instead:

For the user that creates the secret:

Grant ResourceOwner on the Secret resource named conn1
For the user that creates the connector:

Grant ResourceOwner on the Connector resource named conn1
For the principal for the connector itself:

Grant ResourceOwner on the Group resource named connect-conn1
Grant ResourceOwner on the Topic resource named t1
I know that right now "the user that creates the secret", "the user that creates the connector", and "the principal for the connector itself" are the same user (mds), but if we really want to test out the flow that we recommend our users follow, they should all be separate users. The user that creates the secret could still be mds, but the other two should be given separate principals that don't have any other role bindings.

If you don't have extra LDAP users set up and already available, maybe we can just settle for granting all of the above roles on User:mds instead of making it SystemAdmin?"